### ISSUE TITLE
Imgmath Not Producing .svg images

### ISSUE THREAD

**Issue by mia1024** [sphinx-doc/sphinx#3835]
### Problem
- I tried to use imgmath extension. It worked well for producing .png pictures. But when I set `imgmath_img_format` to `svg`, there was no image produced, although the folder `build/html/_image/math` was created. No warning or error was raised either. I  checked and `dvisvgm` is correctly installed and can be invoked from command line. 

#### Procedure to reproduce the problem
```
make html
```

### Reproducible project / your project

 Here are my project source files and the result of running `make html`
[Chemistry Fig Test.zip](https://github.com/sphinx-doc/sphinx/files/1041518/Chemistry.Fig.Test.zip)

### Environment info
- OS: macOS 10.12.3
- Python version: 3.6, 2.7
- Sphinx version: 1.6.2
- TeX distribution with all packages I needed.

**Comment by mia1024** [comment]
Is there any alternative way to use sphinx for some chemistry figures? Especially to include result produced by `mhchem` and `chemfig` TeX packages. I will try to develop an extension for such use if necessary (maybe modify `sphinx.ext.imgmath` slightly?)

**Comment by tk0miya** [comment]
I can generated following image from your example
![2017-06-02 1 55 52](https://cloud.githubusercontent.com/assets/748828/26691043/a9836ca0-4736-11e7-9779-fd12bb20ca0f.png)
(Note: I removed the settings about HTML themes)

To fix this, we need to improve what happened in your environment.
Could you debug the `sphinx.ext.imgmath`?

>Is there any alternative way to use sphinx for some chemistry figures?

Unfortunately, I don't know.

**Comment by jfbu** [comment]
I tried also and it works fine for me like for @tk0miya (tested with Sphinx stable branch, I guess ok with 1.6.2 as well). 

(I also removed the extra html theme, and used classic theme)

1. on `make html`  an `svg` file is correctly produced. I am attaching it here (please remove `.txt` extension). I found it in build folder.

   [e2328fe45e1ecebd77d668faa4e26b6dad0bf8ef.svg.txt](https://github.com/sphinx-doc/sphinx/files/1045582/e2328fe45e1ecebd77d668faa4e26b6dad0bf8ef.svg.txt)

2. before that I had tried `make latexpdf`. I encountered two problems:

    a. you need to add ``'preamble': r'\usepackage{mhchem}',`` to the `latex_elements` dict.

    b. it seems filename with spaces are not supported by pdf makefile. Please modify ``'Chemistry Notebook.tex'`` into something like ``'Chemistry_Notebook.tex'`` in the ``latex_documents`` list.

When adding `mhchem` to my TeX install the dependency `chemgreek` was not installed automatically. I need to install it too. But of course if you are using this package with LaTeX you have the dependency already.

edit: also fine with Sphinx 1.5.6.

**Comment by mia1024** [comment]
So I tried again but still nothing was produced even if I removed the theme. I have no idea why. Here is the screenshot: 
![screen shot 2017-06-01 at 3 22 15 pm](https://cloud.githubusercontent.com/assets/24923261/26696853/1e7f6c1a-46de-11e7-9397-7e7bc19f592c.png)

**Comment by mia1024** [comment]
> Unfortunately, I don't know.

I guess I'll try to develop a new extension after my final exams then. I found sphinx is a really useful tool to take notes. It gives more style control then markdown and easier than LaTeX, yet I can still manually insert html and LaTeX code when necessary. I tried to take notes during class using LaTeX for one term and it was a disaster; I spent all my time formatting the note instead of taking notes.

**Comment by jfbu** [comment]
@pkqxdd I am also on Mac OS X. Perhaps there is some TeX installation issue? You said `dvisvgm` worked from command line, can you check its version, location etc... 

```
$ which -a dvisvgm
/Library/TeX/texbin/dvisvgm

$ dvisvgm --version
dvisvgm (TeX Live) 1.15.1
```
and the `/Library/TeX/texbin/` is from MacTeX install in the past. It is symlinked to actual bin dir in texlive install.

**Comment by mia1024** [comment]
@jfbu I have everything the same as yours. The `.dvi` and `.tex` file  were produced correctly. I tried to convert the `.dvi` file to `svg` manually through command line and it worked fine.

**Comment by jfbu** [comment]
I tested with sphinx 1.6.2 and 1.5.6 on a Linux box with x86_64 GNU/Linux and Ubutun 16.04, both with TL2016 and prerelease TeXLive 2017 and I can not reproduce your issue. For this I commented out in conf.py the theme related things, but you already mentioned this did *not* improve the result for you.

**Comment by jfbu** [comment]
I meant did not improve the result for you

**Comment by jfbu** [comment]
mystery solved. I should have looked more closely at the image you posted at https://github.com/sphinx-doc/sphinx/issues/3835#issuecomment-305594473. But it appeared so small on my screen... You have a space in the absolute path to your project.

As a general rule with all things related to TeX or related binaries such as dvisvgm, never use spaces in filenames or directory names. I reproduced your issue by putting your project Chemistry into ``Folder with space``. Then the `svg` file was produced at same level and was called ``Folder.svg``. And the ``build/html/_images/math`` remained empty.

@tk0miya do you think there is a Sphinx issue in `imgmath` related to spaces in absolute pathes ? personally I suspect more `dvisvgm` or the TeX things such as `kpsewhich` which it may be using.

@pkqxdd: remove spaces in repertory names and you will be allright.

**Comment by mia1024** [comment]
@jfbu Thank you very much. That was exactly the problem. I removed all the spaces in the entire path and now it is working. Just curious, why were png images produced correctly but not svg?

**Comment by jfbu** [comment]
I have not investigated closely so far. It might be a bug in imgmath, or it might simply be that `dvisvgm` is less robust than `dvipng` when spaces appear at some location in paths.

**Comment by jfbu** [comment]
I expect Popen to do the right thing with an option such as `-o`. I.e. possibly use quotes on unix like system or whatever is needed on Windows. But I tried various ways to escape spaces in the `-o` argument of `dvisvgm` with no success so far. It looks currently as limitation of `dvisvgm`. It is bug of Sphinx only if there is a way to use `-o` with spaces for `dvisvgm`. Else it is only bug of Sphinx documentation ;-)

**Comment by jfbu** [comment]
ok, so I think the fix here is to not use the `-o` at all for the dvisvgm call, and to then cp the produced `.svg` file to the destination which is in `outfn` variable. I am not raising an issue yet at https://github.com/mgieseki/dvisvgm for lack of familiarity with its documentation. So far I have not found documentation that `-o` does not like spaces but I tried escaped spaces, or quotes etc... nothing works at all with `dvisvgm` when there are spaces. Typical example:

```
dvisvgm -o "/pathto/Folder\ with\ space/Chemistry/build/html/_images/math/e2328fe45e1ecebd77d668faa4e26b6dad0bf8ef.svg" /var/folders/4y/h4wblhdx08q0j910d0rkq3m80000gn/T/tmpfud923l1/math.dvi
processing of PostScript specials is disabled (Ghostscript not found)
pre-processing DVI file (format 2)
processing page 1
  page size: 267.94pt x 15.2375pt (94.17mm x 5.35538mm)
  page written to ../../Folder\.svg
1 of 1 page converted in 0.310053 seconds
```

**Comment by tk0miya** [comment]
Yes, I also believe this is a bug of dvisvgm.
Good investigation!

**Comment by jfbu** [comment]
Got confirmation at https://github.com/mgieseki/dvisvgm/issues/73 that this was bug of dvisvgm. It is fixed in current release, unfortunately even the not yet released TeXLive 2017 is still using 1.15.1 *on some (older?) architectures*. 

Anyhow we have to handle differently `-o` argument for `dvisvgm`. even if TL2017 had the fix, for supporting older TeX installs.

(Possibly on Windows, the situation is better because MikTeX has a way to keep udpated always, but I don't know which version of the dvisvgm binary it currently distributes)

**Comment by jfbu** [comment]
See [dvisvgm NEWS](https://github.com/mgieseki/dvisvgm/blob/master/NEWS). I don't know which version first fixed the `-o` problem, but anyway, all people with TeXLive 2016 or earlier will still have 1.15.1. And as of dvisvgm-2.0 (2016-08-29) it switched to using C++11 this is reason why even TeXLive 2017 at least on some architectures did not upgrade.

edit: dvisvgm-2.1 is the first bugfix release regarding `-o` problem.

**Comment by tk0miya** [comment]
+0; this is a bug of dvisvgm. Users should be update it. But, on the other hand, it is hard because even TeXLive 2017 does not include it. So it would be nice if we provides some workaround.

**Comment by jfbu** [comment]
@pkqxdd Very soon, TeXLive2017 (and MacTeX)  will be released. As you are on macOS 10.12, this will update dvisvgm to version 2.1.3.

Thus, for you the issue will be fixed with no need of removing space from your ``Class Notes`` folder name.

If you do update to MacTeX 2017, then please tell us if indeed the `dvisvgm` shipped with it fixes the issue.

**Comment by tk0miya** [comment]
Fixed at 32cb60a.
Thank you for reporting!
